src/DotNetWorker.Grpc/Handlers/InvocationHandler.cs

```diff
@@ -0,0 +1,154 @@
+            };
+
+            try
+            {
+                var invocation = new GrpcFunctionInvocation(request);
+                var cancellationTokenSource = new CancellationTokenSource();
+
+                if (!_inflightInvocations.TryAdd(request.InvocationId, cancellationTokenSource))
+                {
+                    // Dispose the CancellationTokenSource if we cannot keep track of it
+                    cancellationTokenSource?.Dispose();
+                }
+
+                IInvocationFeatures invocationFeatures = _invocationFeaturesFactory.Create();
+                invocationFeatures.Set<FunctionInvocation>(invocation);
+                invocationFeatures.Set<IExecutionRetryFeature>(invocation);
+
+                var token = cancellationTokenSource?.Token ?? CancellationToken.None;
+                context = _application.CreateContext(invocationFeatures, token);
+                invocationFeatures.Set<IFunctionBindingsFeature>(new GrpcFunctionBindingsFeature(context, request, _outputBindingsInfoProvider));
+
+                if (_inputConversionFeatureProvider.TryCreate(typeof(DefaultInputConversionFeature), out var conversion))
+                {
+                    invocationFeatures.Set<IInputConversionFeature>(conversion!);
+                }
+
+                await _application.InvokeFunctionAsync(context);
+
+                var functionBindings = context.GetBindings();
+
+                foreach (var binding in functionBindings.OutputBindingData)
+                {
+                    var parameterBinding = new ParameterBinding
+                    {
+                        Name = binding.Key
+                    };
+
+                    if (binding.Value is not null)
+                    {
+                        parameterBinding.Data = await binding.Value.ToRpcAsync(_serializer);
+                    }
+
+                    response.OutputData.Add(parameterBinding);
+                }
+
+                if (functionBindings.InvocationResult != null)
+                {
+                    TypedData? returnVal = await functionBindings.InvocationResult.ToRpcAsync(_serializer);
+                    response.ReturnValue = returnVal;
+                }
+
+                response.Result.Status = StatusResult.Types.Status.Success;
+            }
+            catch (Exception ex)
+            {
+                response.Result.Exception = ex.ToRpcException();
+                response.Result.Status = StatusResult.Types.Status.Failure;
+
+                if (ex.InnerException is TaskCanceledException)
+                {
+                    response.Result.Status = StatusResult.Types.Status.Cancelled;
+                }
+            }
+            finally
+            {
+                _inflightInvocations.TryRemove(request.InvocationId, out var cts);
+
+                if (context is IAsyncDisposable asyncContext)
+                {
+                    await asyncContext.DisposeAsync();
```

Comment:
instead of these explicit disposal calls here, can we just add a `using` statement above to where the context is declared?

Reply:
Same comment as below, will flag this to be addressed outside of this PR #1009